docs/src/modules/components/ThemeContext.js

```diff
@@ -229,15 +228,6 @@ export function ThemeProvider(props) {
         spacing,
       },
       dense ? highDensity : null,
```

Comment:
Actually, instead of removing this we could do:

```
      {
        components: {
          MuiCssBaseline: {
            defaultProps: {
              enableColorScheme: true,
            },
          },
        },
      },
```

With this, you shouldn't need the changes in the previous three files. For this to work you need to fix https://github.com/mui-org/material-ui/pull/29454/files#r742040199 :)